networking_arista/ml2/arista_sec_gp.py

```diff
@@ -197,10 +199,48 @@ class AristaSwitchRPCMixin(object):
 
     def __init__(self, *args, **kwargs):
         super(AristaSwitchRPCMixin, self).__init__()
+        self._statsd = util.STATS
         self._conn_timeout = cfg.CONF.ml2_arista.conn_timeout
         self._verify = cfg.CONF.ml2_arista.verify_ssl
         self._session = kwargs.get('session') or util.make_http_session()
 
+        if self._requests_send_metrics_hook not in \
+                self._session.hooks['response']:
+            self._session.hooks['response'].append(
+                    self._requests_send_metrics_hook)
+
+    def _requests_send_metrics_hook(self, r, *args, **kwargs):
+        """Send request duration to statsd server
+
+        This function should be used as a response hook for the requests
+        library.
+        """
+        # find out callee of _send_eapi_req (but not server())
+        callee_name = "<undefined>"
+        try:
+            eapi_req_found = False
+            for frame in inspect.stack():
+                func_name = frame[3]
+                if not eapi_req_found and func_name == '_send_eapi_req':
+                    eapi_req_found = True
+                elif eapi_req_found and func_name not in ('server', 'wrapped'):
+                    callee_name = func_name
+                    break
+        except Exception:
+            pass
+
+        # parse url to get hostname later on
+        parsed_url = urllib2.urlparse.urlparse(r.url)
```

Comment:
Could you use `six.moves.urllib.parse` instead?
Best we try to keep the code python3 compatible.